### ISSUE TITLE
Add signature fields to compressed tx form

### ISSUE THREAD

**Issue by ChihChengLiang** [thehubbleproject/RedditHubble#113]
As we concluded in the call Monday, we want to add a signature field to the transaction fields of the compressed form.

We'll use Transfer as example

## Before

Currently, the struct Tx.Transfer looks like

```
    struct Transfer {
        uint256 senderID;
        uint256 receiverID;
        uint256 amount;
        uint256 nonce;
    }
```
And it will be compressed to `[sender_state_id<4>|receiver_state_id<4>|amount<4>|nonce<4>]`

This means for ECDSA version we had to commit signatures separately to `submitBatch`, and we need to merklize/hashchain the signatures together and add a field in the Batch.

## After

We add the signature field in Tx.Transfer 

```diff
    struct Transfer {
        uint256 senderID;
        uint256 receiverID;
        uint256 amount;
        uint256 nonce;
+     bytes signature;
    }
```
And it will be compressed to `[sender_state_id<4>|receiver_state_id<4>|amount<4>|nonce<4>|signature<64>]`

So the signature will be committed with the transaction itself. No extra merkle/hashchain/field in Batch.

**Comment by ChihChengLiang** [comment]
Being Implemented in #111

**Comment by ChihChengLiang** [comment]
Fixed in #111 and #114
